exp/controllers/rosamachinepool_controller.go

```diff
@@ -307,13 +308,22 @@ func (r *ROSAMachinePoolReconciler) reconcileMachinePoolVersion(machinePoolScope
 	}
 
 	clusterID := *machinePoolScope.ControlPlane.Status.ID
-	scheduledUpgrade, err := rosaClient.CheckNodePoolExistingScheduledUpgrade(clusterID, nodePool)
+	_, scheduledUpgrade, err := ocmClient.GetHypershiftNodePoolUpgrade(clusterID, machinePoolScope.ControlPlane.Spec.RosaClusterName, nodePool.ID())
 	if err != nil {
 		return fmt.Errorf("failed to get existing scheduled upgrades: %w", err)
 	}
 
 	if scheduledUpgrade == nil {
-		scheduledUpgrade, err = rosaClient.ScheduleNodePoolUpgrade(clusterID, nodePool, version, time.Now())
+		rawVersion := ocm.GetRawVersionId(version)
+		policy, err := ocmClient.BuildNodeUpgradePolicy(rawVersion, nodePool.ID(), ocm.UpgradeScheduling{
+			AutomaticUpgrades: false,
+			NextRun:           time.Now(),
```

Comment:
`time.now()` doesn't work as explained in the comment of the original function.

Reply:
Please explain what you mean.